src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    @Override
+    public Class<NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUIPanel> getPayloadGeneratorUIPanelClass() {
+        return NumberPayloadGeneratorUIPanel.class;
+    }
+
+    @Override
+    public PayloadGeneratorUIPanel<DefaultPayload, NumberPayloadGenerator, NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUI> createPanel() {
+        return new NumberPayloadGeneratorUIPanel();
+    }
+
+    public static class NumberPayloadGeneratorUI implements PayloadGeneratorUI<DefaultPayload, NumberPayloadGenerator> {
+
+        NumberPayloadGenerator numberPayloadGenerator;
+
+        public NumberPayloadGeneratorUI(NumberPayloadGenerator generator) {
+            this.numberPayloadGenerator = generator;
+        }
+
+        @Override
+        public Class<? extends NumberPayloadGenerator> getPayloadGeneratorClass() {
+            return NumberPayloadGenerator.class;
+        }
+
+        @Override
+        public String getName() {
+            return PAYLOAD_GENERATOR_NAME;
+        }
+
+        @Override
+        public String getDescription() {
+            return PAYLOAD_GENERATOR_DESC;
+        }
+
+        @Override
+        public long getNumberOfPayloads() {
+            try {
+                return numberPayloadGenerator.getNumberOfPayloads();
+            } catch (Exception e) {
+                LOGGER.warn("Failed to obtain number of payloads", e);
+            }
+            return PayloadGenerator.UNKNOWN_NUMBER_OF_PAYLOADS;
+        }
+
+        @Override
+        public NumberPayloadGenerator getPayloadGenerator() {
+            return numberPayloadGenerator;
+        }
+
+        @Override
+        public PayloadGeneratorUI<DefaultPayload, NumberPayloadGenerator> copy() {
+            return this;
+        }
+
+    }
+
+    public static class NumberPayloadGeneratorUIPanel extends AbstractPersistentPayloadGeneratorUIPanel<DefaultPayload, NumberPayloadGenerator, NumberPayloadGeneratorUI> {
+
+        private static final String PAYLOADS_PREVIEW_FIELD_LABEL = Constant.messages.getString("fuzz.payloads.generator.numbers.payloadsPreview.label");
+        private static final String PAYLOADS_PREVIEW_GENERATE_FIELD_LABEL = Constant.messages.getString("fuzz.payloads.generator.numbers.payloadsPreviewGenerate.label");
+        private static final String PAYLOADS_FROM_LABEL = Constant.messages.getString("fuzz.payloads.generator.numbers.from.label");
+        private static final String PAYLOADS_TO_LABEL = Constant.messages.getString("fuzz.payloads.generator.numbers.to.label");
+        private static final String PAYLOADS_INCREMENT_LABEL = Constant.messages.getString("fuzz.payloads.generator.numbers.increment.label");
+
+        private static final int MAX_NUMBER_PAYLOADS_PREVIEW = 250;
+
+        private JPanel fieldsPanel;
+        private JTextArea payloadsPreviewTextArea;
+        private JButton payloadsPreviewGenerateButton;
+        private JTextField fromField, toField, stepField;
```

Comment:
- Consider adding each variable in its own line, it's easier to read/change.
 - It should use JSpinner instead of JTextField, maybe ZapNumberSpinner? (Unless the Integer limits are not within the target range.)